src/ica_api.c

```diff
@@ -1017,7 +1017,11 @@ unsigned int ica_rsa_mod_expo(ica_adapter_handle_t adapter_handle,
 		rc = ica_fallbacks_enabled ?
 			rsa_mod_expo_sw(&rb) : ENODEV;
 	else {
-		rc = ioctl(adapter_handle, ICARSAMODEXPO, &rb);
+		if (any_card_online)
+			rc = ioctl(adapter_handle, ICARSAMODEXPO, &rb);
+		else
+			rc = 1;
```

Comment:
Is there a reason why you return "1" here, an not ENODEV or similar?

Reply:
I agree, ENODEV is better here. Code changed.